TESTS/mbedmicro-rtos-mbed/timer/main.cpp

```diff
@@ -0,0 +1,872 @@
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+#include "rtos.h"
+#include "timer.h"
+#include "hal/us_ticker_api.h"
+
+using namespace utest::v1;
+
+#define US_PER_SEC 1000000
+
+/* NOTE: NRF51_DK platform has 32kHz timer clock which
+ *       gives resolution of 31.25us/clock tick.
+ *       Therefore for this board we need a huge delta
+ *       for the timing tests. */
+
+/* Macro to define delta based on system clock frequency.
+ * Assume that min. delta value is equal to 5 us.
+ *
+ * Calculate number of us per system tick. Then multiple calculated value by
+ * 10 and add 5. Truncate calculated delta to decimal value by casting to int.
+ *
+ * For 1 MHz clock delta is equal to 15 us.
+ *
+ * EXAMPLE K64F
+ * system clock frequency = 120 000 000 Hz (120 MHz) => 1 tick = ~0.083 us.
+ * delta = truncate(0.083 * 10 + 5) = truncate(5.83) = 5 us.
+ *
+ * EXAMPLE NRF51_DK
+ * system clock frequency = 32 678 Hz (~32 kHz) => 1 tick = 31.25 us.
+ * delta = truncate(31.25 * 10 + 5) = truncate(317.5) = 317 us.
+ */
+#define DELTA_SYS_CLK_US ((int)((1.0f / ((float)osKernelGetSysTimerFreq() / US_PER_SEC) ) * 10 + 5))
```

Comment:
It is possible, but then it would have to be a very large value (e.g. 500 us), to be able to successfully execute test on each board. It looks like the better option is to define delta value based on board clock frequency.